src/shared/bridge.js

```diff
@@ -78,46 +76,54 @@ export default class Bridge {
    * callback when all results are collected.
    *
    * @param {string} method - Name of remote method to call.
-   * @param {any[]} args - Arguments to method.
+   * @param {any[]} args - Arguments to method. Final argument is an optional
+   *   callback with this type: `(error: string|Error, ...result: any[]) => void`
    * @return {Promise<any[]>} - Array of results, one per connected frame
    */
   call(method, ...args) {
     let cb;
-    if (typeof args[args.length - 1] === 'function') {
-      cb = args[args.length - 1];
+    const finalArg = args[args.length - 1];
+    if (typeof finalArg === 'function') {
+      cb = finalArg;
       args = args.slice(0, -1);
     }
 
-    const _makeDestroyFn = c => {
+    /** @param {RPC} channel */
+    const _makeDestroyFn = channel => {
       return error => {
-        c.destroy();
-        this.links = Array.from(this.links)
-          .filter(l => l.channel !== c)
-          .map(l => l);
+        channel.destroy();
+        this.links = this.links.filter(
+          registeredChannel => registeredChannel !== channel
+        );
         throw error;
       };
     };
 
-    const promises = this.links.map(l => {
-      const p = new Promise((resolve, reject) => {
+    const promises = this.links.map(channel => {
+      const promise = new Promise((resolve, reject) => {
         const timeout = setTimeout(() => resolve(null), 1000);
+        // This try/catch is only used on tests, to exercise the `_makeDestroyFn`
+        // function. In real life, the execution of the channel.call's callback will
```

Comment:
The `catch` block is not invoked if `call`'s _callback_ throws but if `call` itself throws. Looking at the current implementation I think that is only possible if `postMessage` throws.

Reply:
Yes, you are correct.